Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -142,35 +147,34 @@ public static IEnumerable<string> FindPluginAssemblies(string pluginFolder, bool
                 {
                     continue;
                 }
-
-                if (assemblyPath.Contains("Extensions"))
+                
+                string extensionFolder = Path.GetDirectoryName(assemblyPath);
+                if (Directory.Exists(extensionFolder))
                 {
-                    string extensionFolder = Path.GetDirectoryName(assemblyPath);
-                    var validationResult = validator.ValidateObject(extensionFolder);
-
-                    if (!validationResult.Any())
+                    string[] assemblyMetadataPath = Directory.GetFiles(
+                        extensionFolder, AssemblyMetadataSearchString, SearchOption.AllDirectories);
+                        
+                    if (assemblyMetadataPath.Length>0 && IsExtensionInAssemblyModelType(assemblyMetadataPath[0]))
                     {
-                        yield return assembly.Location;
+                        var validator = GetValidator();

```

Comment:
For clarity, please rename this method to `GetExtensionValidator` since the validations only apply to extension assemblies and _definitely_ not the `else` condition for the profile assemblies.

Reply:
yes ,  I updated . Thanks